### PR TITLE
[CodeClean] possible mem leak case

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Code clean, update util function to free allocated mem block and fix mem leak case.

### CODE REVIEW

File: gst/datarepo/gstdatareposrc.c [hunk 1]
```diff
@@ -1041,6 +1041,7 @@ gst_data_repo_src_read_others (GstDataRepoSrc * src, GstBuffer ** buffer)
 
   if (!gst_memory_map (mem, &info, GST_MAP_WRITE)) {
     GST_ERROR_OBJECT (src, "Could not map GstMemory");
+    gst_memory_unref (mem);
```
Comment (songgot) [inline gst/datarepo/gstdatareposrc.c:1044]: Thank you for checking with your hawk eyes. 🦅

File: gst/nnstreamer/elements/gsttensor_reposrc.c [hunk 2]
```diff
@@ -303,17 +303,16 @@ gst_tensor_reposrc_gen_dummy_buffer (GstTensorRepoSrc * self)
     _info = gst_tensors_info_get_nth_info (&self->config.info, i);
     size = gst_tensor_info_get_size (_info);
     mem = gst_allocator_alloc (NULL, size, NULL);
+    gst_tensor_buffer_append_memory (buf, mem, _info);
 
     if (!gst_memory_map (mem, &map, GST_MAP_WRITE)) {
-      gst_allocator_free (NULL, mem);
       gst_buffer_unref (buf);
-      ml_logf ("Cannot mep gst memory (tensor-repo-src)\n");
+      ml_logf ("Cannot map gst memory (tensor-repo-src).");
```
Comment (songgot) [inline gst/nnstreamer/elements/gsttensor_reposrc.c:310]: Maybe, ```gst_memory_unref (mem)``` will be done inside gst_buffer_unref(buf), right?

**Review by songgot — APPROVED** [review]
LGTM 😄

File: gst/nnstreamer/elements/gsttensor_reposrc.c (hunk 2 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/elements/gsttensor_reposrc.c:310]: right :)

**Review by anyj0527 — APPROVED** [review]
LGTM✔

**Pull request merged**